pylint/checkers/strings.py

```diff
@@ -834,16 +834,16 @@ def check_for_concatenated_strings(
 
     def process_string_token(self, token: str, start_row: int, start_col: int) -> None:
         quote_char = None
-        index = None
-        for index, char in enumerate(token):
+        _index = None
```

Comment:
Why are we even initialising `_index` here?

Reply:
We're implicitly using the enumerate to have the latest index value later. We could refactor to something clearer, not sure about the perf implication though.

Reply:
You but we check `quote_char` so `_index` is always a value on L845 and L846. We don't need the `_index = None` as far as I can see.

Reply:
This would be a syntax error in python as _index do not leave the for loop scope if it's not defined before.

Reply:
```console
❯ python
Python 3.11.0b4 (main, Jul 12 2022, 11:43:04) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> for index in range(1, 2):
...     print(index)
...
1
>>> print(index)
1
```

There is actually a `pylint` issue about this behaviour as it is both dangerous and useful 😄

Reply:
```python
>>> for i, x in enumerate([]):
...     print(i, x)
... print(i)
  File "<stdin>", line 3
    print(i)
    ^^^^^
SyntaxError: invalid syntax
```

Reply:
But this can't happen due to L842 right?

Reply:
Right, I'll fix that.

Reply:
We get 
```
************* Module pylint.checkers.strings

pylint/checkers/strings.py:844:24: W0631: Using possibly undefined loop variable '_index' (undefined-loop-variable)

pylint/checkers/strings.py:845:29: W0631: Using possibly undefined loop variable '_index' (undefined-loop-variable)

```
Which is probably the reason why we had the index definition in the first place.

Reply:
Ah, let's disable `pylint` there then. I prefer disabling tools that don't understand Python instead of adding more Python to make tools understand it.

Reply:
266c0f545